boot/freeldr/freeldr/arch/drivers/hwide.c

```diff
@@ -461,11 +461,52 @@ AtapiCapacityDetect(
 
         *SectorSize = (AtapiCapacity[4] << 24) | (AtapiCapacity[5] << 16) |
                       (AtapiCapacity[6] << 8) | AtapiCapacity[7];
+
+        /*
+         * The file system driver should use only the ARC interface,
+         * that doesn't allow for retrieving the disk geometry
+         * Reset to default value
+         */
+        if(*SectorSize != 0)
+            *SectorSize = 2048;
     }
     else
     {
         *TotalSectors = 0;
         *SectorSize = 0;
+
+        AtapiPrintSenseData(DeviceUnit);
+    }
+}
+
+static
+VOID
+AtapiPrintSenseData(IN PDEVICE_UNIT DeviceUnit)
+{
+    UCHAR AtapiPacket[16];
+    SENSE_DATA SenseData;
+    BOOLEAN Success;
+
+    /* Send the SCSI REQUEST SENSE command */
+    RtlZeroMemory(&AtapiPacket, sizeof(AtapiPacket));
+    RtlZeroMemory(&SenseData, SENSE_BUFFER_SIZE);
```

Comment:
`C_ASSERT(sizeof(SenseData) >= SENSE_BUFFER_SIZE)` please.

Reply:
Is this assert really needed? SENSE_DATA and SENSE_BUFFER_SIZE were taken from DDK: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/ddk/scsi.h;hb=e410a12242a50e850846058ef5c53652340c1724#l524

Reply:
yes for correctness the `SENSE_BUFFER_SIZE` should have been defined as being equal to `sizeof(SENSE_DATA)`.